pkg/ctl/scale/nodegroup.go

```diff
@@ -57,10 +56,38 @@ func scaleNodeGroupWithRunFunc(cmd *cmdutils.Cmd, runFunc func(cmd *cmdutils.Cmd
 }
 
 func doScaleNodeGroup(cmd *cmdutils.Cmd, ng *api.NodeGroupBase) error {
-	if err := cmdutils.NewScaleNodeGroupLoader(cmd, ng).Load(); err != nil {
-		return err
+	if ng.Name == "" && cmd.NameArg == "" {
+		if err := cmdutils.NewScaleAllNodeGroupLoader(cmd).Load(); err != nil {
+			return err
+		}
+		if err := scaleAllNodegroups(cmd); err != nil {
+			return err
+		}
+	} else {
+		if err := cmdutils.NewScaleNodeGroupLoader(cmd, ng).Load(); err != nil {
+			return err
+		}
+		if err := scaleNodegroup(cmd, ng); err != nil {
+			return err
+		}
 	}
+	return nil
```

Comment:
nice. Maybe rather than `if else` do a `if` with a return at the end? Less brain power required :smile: 

```suggestion
	if ng.Name == "" && cmd.NameArg == "" {
		if err := cmdutils.NewScaleAllNodeGroupLoader(cmd).Load(); err != nil {
			return err
		}
		return scaleAllNodegroups(cmd);
	}
	
        if err := cmdutils.NewScaleNodeGroupLoader(cmd, ng).Load(); err != nil {
        	return err
        }
        return scaleNodegroup(cmd, ng)

```